src/java/org/apache/cassandra/service/RowDataResolver.java

```diff
@@ -109,6 +110,11 @@ public Row resolve() throws DigestMismatchException
      */
     public static List<AsyncOneResponse> scheduleRepairs(ColumnFamily resolved, String keyspaceName, DecoratedKey key, List<ColumnFamily> versions, List<InetAddress> endpoints)
```

Comment:
I think there are other parts of the code that do async read repairs along the write path: https://github.com/palantir/cassandra/blob/e301a8dfa26197f9585e11011b9a3649e5858f99/src/java/org/apache/cassandra/service/ReadCallback.java#L272. But those are async as far as I can tell and don't seem to be leveraged in the read path. So maybe not worth tackling for now

Reply:
Yeah, I'm not convinced that disabling async read repairs is even desirable at this point

Reply:
From what I can tell, I think this actually goes back into `RowDataResolver::resolve` code path and from a quick lookup in the code `MessagingService.Verb.READ_REPAIR` is only leveraged in `scheduleRepairs`. So we would effectively be disabling all read repairs on reads or writes (which makes me lean towards you and Lyuben's suggestion to be incremental more, considering we seem to already be throwing around a bunch of async read repairs when we write)